src/cryptonote_basic/account.cpp

```diff
@@ -60,7 +63,71 @@ DISABLE_VS_WARNINGS(4244 4345)
     m_device = &hwdev;
     MCDEBUG("device", "account_keys::set_device device type: "<<typeid(hwdev).name());
   }
+  //-----------------------------------------------------------------
+  static void derive_key(const crypto::chacha_key &base_key, crypto::chacha_key &key)
+  {
+    static_assert(sizeof(base_key) == sizeof(crypto::hash), "chacha key and hash should be the same size");
+    char data[sizeof(base_key) + 1];
```

Comment:
Maybe this is a good place to use `epee::mlocked<tools::scrubbed_arr<char,sizeof(base_key)+1>>` as in `wallet2::setup_keys`?